packages/material-ui/src/TableRow/TableRow.js

```diff
@@ -50,27 +56,33 @@ function TableRow(props) {
     classes,
     className: classNameProp,
     component: Component,
+    dense,
     hover,
     selected,
     ...other
   } = props;
 
   return (
-    <Tablelvl2Context.Consumer>
-      {tablelvl2 => {
-        const className = clsx(
-          classes.root,
-          {
-            [classes.head]: tablelvl2 && tablelvl2.variant === 'head',
-            [classes.footer]: tablelvl2 && tablelvl2.variant === 'footer',
-            [classes.hover]: hover,
-            [classes.selected]: selected,
-          },
-          classNameProp,
-        );
-        return <Component className={className} {...other} />;
-      }}
-    </Tablelvl2Context.Consumer>
+    <TableContext.Consumer>
+      {table => (
+        <Tablelvl2Context.Consumer>
```

Comment:
It's probably time to use the hook API.

Reply:
Yup, this is far too much nesting.

Reply:
Which React version is material-ui@next targeting? Maybe we could also drop `this.useMemo`.

Reply:
16.8, so you can use `useMemo`

Reply:
@mbrookes So… we could use `React.useMemo` instead of the `this.useMemo` helper everywhere?